apps/client/src/features/table/table-settings/TableSettings.tsx

```diff
@@ -0,0 +1,74 @@
+import { Button, Checkbox, Switch } from '@chakra-ui/react';
+import { Column } from '@tanstack/react-table';
+import { OntimeRundownEntry } from 'ontime-types';
+
+import { useCuesheetSettings } from '../store/CuesheetSettings';
+
+import style from './TableSettings.module.scss';
+
+// reusable button styles
+const buttonProps = {
+  size: 'sm',
+  variant: 'ontime-subtle',
+};
+
+interface CuesheetSettingsProps {
+  columns: Column<OntimeRundownEntry, unknown>[];
+  handleResetResizing: () => void;
+  handleResetReordering: () => void;
+  handleClearToggles: () => void;
+}
+
+export function CuesheetSettings(props: CuesheetSettingsProps) {
+  const { columns, handleResetResizing, handleResetReordering, handleClearToggles } = props;
+  const showDelayBlock = useCuesheetSettings((state) => state.showDelayBlock);
+  const toggleDelayVisibility = useCuesheetSettings((state) => state.toggleDelayVisibility);
+  const showPrevious = useCuesheetSettings((state) => state.showPrevious);
+  const togglePreviousVisibility = useCuesheetSettings((state) => state.togglePreviousVisibility);
+
+  return (
+    <div className={style.tableSettings}>
+      <div className={style.leftPanel}>
+        <div className={style.sectionTitle}>Toggle field visibility</div>
+        <div className={style.options}>
```

Comment:
##### For the checkboxes
If you use a `grid` (and `flex`, too), you can substitute the `padding` for `gap`. 
<img width="329" alt="Screenshot 2023-07-12 at 1 33 00 PM" src="https://github.com/cpvalente/ontime/assets/58940073/c9c836f2-cdd5-4661-90cb-fb3d081c2e4f">

##### For the toggles
I don't think this padding looks nice:
<img width="395" alt="Screenshot 2023-07-12 at 1 32 33 PM" src="https://github.com/cpvalente/ontime/assets/58940073/660cef6d-f326-4963-8c68-4f3f93f2efb0">

Reply:
Good observation, thank you!

Do you think there is anything in terms of style that we can do here to make this more expandable?
As it is, I fear this can grow into a complicated UX

Reply:
I think it's fine if they're evened out. 
If you want to get fancy, I think organizing the options as a list in a modal could be an alternative. I'm not sure, but if theoretically, there's a hundred options, that'll be the way to go.